src/gwc/src/main/java/org/geoserver/gwc/dispatch/GeoServerSecurityFilter.java

```diff
@@ -47,4 +52,31 @@ public void checkSecurity(TileLayer layer, BoundingBox extent, SRS srs)
             }
         }
     }
+
+    // copied from org.geoserver.web.spring.security.GeoServerSession.isAdmin()
+    @Override
```

Comment:
this annotation causes a build failure
```
Error: 0,529 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project gs-gwc: Compilation failure
Error: 0,529 [ERROR] /home/runner/work/geoserver/geoserver/src/gwc/src/main/java/org/geoserver/gwc/dispatch/GeoServerSecurityFilter.java:[57,5] method does not override or implement a method from a supertype
Error: 0,529 [ERROR] -> [Help 1]
```

Reply:
This PR is dependent on a GeoWebCache PR and can't build properly without it. Removing the annotation will cause a QA failure once the GWC PR is merged.

Reply:
right; I missed that; you'll have to revert my commit in that case, in the meantime I'll mark this as "draft"